src/App.js

```diff
@@ -79,7 +75,9 @@ export default function App() {
                         fontSize: 14,
                     }}
                 >
-                    {visitorHistory}
+                    {visitorHistory.visits
+                        ? JSON.stringify(visitorHistory.visits, null, 4)
+                        : null}
```

Comment:
It doesn't make much sense to render empty h3 and pre tags, so what's usually done in react is this:
```
{visitorHistory && visitorHistory.visits &&
<h3>
   Received history of {visitorHistory.visits.length} visits:
</h3>
<pre
  style={{
  fontFamily: "inherit",
  fontSize: 14,
  }}
>
  {visitorHistory}
  {JSON.stringify(visitorHistory.visits, null, 4)}
</pre>}
```
(JSX will only be rendered if visits is not null